apps/web/playwright/onboarding.e2e.ts

```diff
@@ -9,75 +11,89 @@ test.afterEach(({ users }) => users.deleteAll());
+            await expect(page).toHaveURL(/.*setup-availability/);
+          } else {
+            await expect(page).toHaveURL(/.*connected-calendar/);
+          }
+
+          const userComplete = await user.self();
+          expect(userComplete.name).toBe("new user 2");
+        });
+
+        if (identityProvider !== IdentityProvider.GOOGLE) {
+          await test.step("step 2 - Connected Calendar", async () => {
+            const isDisabled = await page.locator("button[data-testid=save-calendar-button]").isDisabled();
+            await expect(isDisabled).toBe(true);
+            // tests skip button, we don't want to test entire flow.
+            await page.locator("button[data-testid=skip-step]").click();
+
+            await expect(page).toHaveURL(/.*connected-video/);
+          });
+
+          await test.step("step 3 - Connected Video", async () => {
+            const isDisabled = await page.locator("button[data-testid=save-video-button]").isDisabled();
+            await expect(isDisabled).toBe(true);
+            // tests skip button, we don't want to test entire flow.
+            await page.locator("button[data-testid=skip-step]").click();
+
+            await expect(page).toHaveURL(/.*setup-availability/);
+          });
+        }
+
+        await test.step("step 4 - Setup Availability", async () => {
+          const isDisabled = await page.locator("button[data-testid=save-availability]").isDisabled();
+          await expect(isDisabled).toBe(false);
+          // same here, skip this step.
+          await page.locator("button[data-testid=save-availability]").click();
+
+          await expect(page).toHaveURL(/.*user-profile/);
+        });
+
+        await test.step("step 5- User Profile", async () => {
+          await page.locator("button[type=submit]").click();
+
+          // should redirect to /event-types after onboarding
+          await page.waitForURL("/event-types");
+
+          const userComplete = await user.self();
+
+          expect(userComplete.bio?.replace("<p><br></p>", "").length).toBe(0);
+        });
       });
+    };
 
-      await test.step("step 4", async () => {
-        const isDisabled = await page.locator("button[data-testid=save-availability]").isDisabled();
-        await expect(isDisabled).toBe(false);
-        // same here, skip this step.
-        await page.locator("button[data-testid=save-availability]").click();
-
-        await expect(page).toHaveURL(/.*user-profile/);
-      });
-
-      await test.step("step 5", async () => {
-        await page.locator("button[type=submit]").click();
-
-        // should redirect to /event-types after onboarding
-        await page.waitForURL("/event-types");
-
-        const userComplete = await user.self();
-
-        expect(userComplete.bio?.replace("<p><br></p>", "").length).toBe(0);
-      });
-    });
+    testOnboarding(IdentityProvider.GOOGLE);
+    testOnboarding(IdentityProvider.CAL);
+    testOnboarding(IdentityProvider.SAML);
```

Comment:
q: do we need to await Promise.all? or no need?

Reply:
Good Q, let me play with it